source/plugins/ruby/in_containerinventory.rb

```diff
@@ -82,6 +83,15 @@ def enumerate
                   if hostName.empty? && !containerRecord["Computer"].empty?
                     hostName = containerRecord["Computer"]
                   end
+                  if addonTokenAdapterImageTag.empty? && ExtensionUtils.isAADMSIAuthMode()
+                     if !containerRecord["ElementName"].nil? && !containerRecord["ElementName"].empty? &&
```

Comment:
I also disagree with this approach. Much neater way to do this would be, pass this image as an env variable from yaml. Define a helm variable for sidecar container, and use that for the spec.container.image as well pass the same as env variable to be picked up by our telemetry